### PR TITLE
Routine updates to get JDK11 build compatibility

### PR DESCRIPTION
**Pull request by Vlatombe**
* Update parent POM
* Use BOM
* Remove explicit versions

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,76 +42,75 @@
     <properties>
         <revision>1.4</revision>
         <changelist>-SNAPSHOT</changelist>
-        <jenkins.version>2.60.3</jenkins.version>
+        <bom>2.204.x</bom>
+        <jenkins.version>2.204.6</jenkins.version>
```
Comment (alecharp) [inline pom.xml:46]:
```suggestion
        <bom>2.289.x</bom>
        <jenkins.version>2.289.3</jenkins.version>
```

File: pom.xml [hunk 2]
```diff
@@ -42,76 +42,75 @@
...
         <java.level>8</java.level>
-        <workflow-support-plugin.version>2.13</workflow-support-plugin.version>
-        <workflow-cps-plugin.version>2.28</workflow-cps-plugin.version>
     </properties>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-${bom}</artifactId>
+                <version>18</version>
```
Comment (alecharp) [inline pom.xml:54]:
```suggestion
                <version>1036.v9f5a1aba8fab</version>
```

File: pom.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>3.50</version>
+        <version>4.29</version>
```
Comment (alecharp) [inline pom.xml:7]:
```suggestion
        <version>4.31</version>
```

**Review by alecharp — APPROVED** [review]
Suggested updated versions to use but it could be merge as is and let dependabot update everything shortly.

**Comment by Vlatombe** [comment]
Superceded by #23

**Pull request closed without merging**